src/rpc/methods/eth.rs

```diff
@@ -157,6 +157,16 @@ lotus_json_with_self!(Int64);
 pub struct Hash(#[schemars(with = "String")] pub ethereum_types::H256);
 
 impl Hash {
+    // Should ONLY be used for blocks and Filecoin messages. Eth transactions expect a different hashing scheme.
+    pub fn to_cid(&self) -> cid::Cid {
+        use cid::multihash::MultihashDigest;
+
+        let mh = cid::multihash::Code::Blake2b256
+            .wrap(self.0.as_bytes())
+            .expect("should not fail");
```

Comment:
Is there a possibility of this panicking on bogus input provided by the RPC caller? If so, I'd avoid panicking and return a `Result`.

Reply:
It's not possible to fail here, given some malicious RPC payload. By construction an `H256` is made of an `[u8; 32]` and can fit into a `Multihash<64>`.

There is some quickcheck test for the roundtrip.